Libraries/LibWeb/WebAssembly/Memory.cpp

```diff
@@ -98,20 +109,32 @@ WebIDL::ExceptionOr<GC::Ref<JS::ArrayBuffer>> Memory::buffer() const
     auto& realm = *vm.current_realm();
 
     if (!m_buffer)
-        m_buffer = TRY(create_a_memory_buffer(vm, realm, m_address));
+        m_buffer = TRY(create_a_memory_buffer(vm, realm, m_address, m_shared));
 
     return GC::Ref(*m_buffer);
 }
 
 // https://webassembly.github.io/spec/js-api/#create-a-memory-buffer
```

Comment:
this spec link is outdated. 

https://webassembly.github.io/spec/js-api/#create-a-memory-buffer

It seems the JS API has been updated since this was done?

It looks like it was updated to support resizable array buffer.

https://webassembly.github.io/spec/js-api/#create-a-fixed-length-memory-buffer

Reply:
Yes, it looks like the API was updated.

I've put references to the newer spec where applicable. I would like to have a deeper look into the diff between the specs and update the references in the next PR.